### ISSUE TITLE
Need help for replicating Low Resource Track.

### ISSUE THREAD

**Issue by sai-prasanna** [kakaobrain/helo_word#5]
I followed the instructions for the low-resource track. After the DAE step, I get 0.33 score with evaluate.py on the best model.
But on training with the 3k dev set and evaluating, the score becomes very low. I tried increasing the number of epochs, lowering learning rates. But I am unable to even retain the F score from the DAE pre-training.
Need help to replicate the train step.

**Comment by hammouse** [comment]
We used 5e-4 learning rate for the DAE pre-training. 
We also did not reset the optimizer when we start the training step.

**Comment by sai-prasanna** [comment]
@hammouse As a clarification, what is the single model F-score, final validation loss you get for low resource track after training on 3k dev set? Can you share the training log?

I didn't change any learning rate or other hyper parameters, just used the track 3 training after pretraining commands you have provided.

My loss starts at 15.0 and perplexity is very high and as the training dataset is small (just 3k) it ends fairly quickly. And all the trained epoch checkpoints on evaluation, result is very poor since model generates incoherent text after training on 3k.

**Comment by sai-prasanna** [comment]
@hammouse  Can you release a trained model for the third track, or just the logs for training ?

**Comment by hammouse** [comment]
At the end of the pre-training, the validation loss reached 2.976.
After the training, the validation loss reached 2.705.

**Comment by JesseWong333** [comment]
@sai-prasanna  @hammouse 
I really appreciate the work that @hammouse have done, which save me lots of time.  
When I followed the instructions for track 1, I found 
`no existing checkpoint found /media/Data/XXX/XXX/track1/ckpt/pretrain-base-lr0.0005-dr0.3_train-base-lr0.0001-dr0.3/./track1/ckpt/pretrain-base-lr0.0005-dr0.3_train-base-lr0.0001-dr0.3/checkpoint_3_150000.pt`
It seemed that Fairseq did not found the retore checkpoint, and start to train from scratch.

**Comment by 13145314593** [comment]
> We used 5e-4 learning rate for the DAE pre-training.
> We also did not reset the optimizer when we start the training step.

My network is jammed, could you share the contents in the data folder  or bpe-model with me?

**Comment by niucheney** [comment]
I had the similar problem.  There is a bug in the script  "train.py" when I submit the job to the cluster server. Look at the following code in the funciton of "train":
            finetune_ckpt = os.path.basename(utils.change_ckpt_dir(restore_ckpt, ckpt_dir)) 
            os.system(f"cp {restore_ckpt} {ckpt_dir}")
Because os.path.basename only get the basename of "restore_ckpt".  As a result, the following command cannot cp restore_ckpt into a new directory which is ckpt_dir.

**Comment by niucheney** [comment]
Because there is a bug in train.py. I fixed the bug and had replicated the experiment.  
In "def train " function,  I fixed the bug as follows.
`def train(databin_path, model_config, ckpt_dir, restore_ckpt, ngpu):
    xxx

    if restore_ckpt is not None:
        finetune_ckpt =util.change_ckpt_dir(restore_ckpt, ckpt_dir)
        logging.info(f"[Train] copy the ckpt {restore_ckpt} into {finetune_ckpt}")
        os.system(f"cp {restore_ckpt} {finetune_ckpt}")
        finetune_ckpt = os.path.basename(util.change_ckpt_dir(restore_ckpt, ckpt_dir))
        prompt += f"--restore-file {finetune_ckpt} "
    os.system(prompt)`
